File: src/main/java/com/example/cqrs/command/dto/CreateAccountRequest.java
package com.example.cqrs.command.dto;

import lombok.*;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class CreateAccountRequest {

    private String accountId;
    private String userId;

    // factory method
    public static CreateAccountRequest of(String accountId, String userId) {
        return new CreateAccountRequest(accountId, userId);
    }

}

File: src/main/java/com/example/cqrs/command/dto/CreateAccountRequest.kt
package com.example.cqrs.command.dto

data class CreateAccountRequest(
    val accountId: String,
    val userId: String,
) {
    companion object {
        fun of(accountId: String, userId: String): CreateAccountRequest {
            return CreateAccountRequest(accountId, userId)
        }
    }
}
